File: src/features/fees/types/feeRule.ts
export type FeeRuleMarket = 'crypto' | 'forex' | 'commodities' | 'indices' | 'stocks'

export type FeeRule = {
  id: string
  groupId: string
  groupName: string
  /** Empty / undefined means all symbols in group */
  symbol?: string | null
  market?: FeeRuleMarket | null
  feePercent: number
  minFee: number
  maxFee?: number | null
  status: 'active' | 'disabled'
  notes?: string | null
  createdAt: string
  updatedAt: string
  updatedBy?: string | null
  createdByUserId?: string | null
  createdByEmail?: string | null
}

export interface ListFeeRulesParams {
  groupId?: string
  symbol?: string
  status?: string
  page?: number
  pageSize?: number
}

export interface ListFeeRulesResponse {
  items: FeeRule[]
  page: number
  page_size: number
  total: number
}

export interface CreateFeeRulePayload {
  groupId: string
  symbol?: string | null
  market?: FeeRuleMarket | null
  feePercent: number
  minFee: number
  maxFee?: number | null
  status: 'active' | 'disabled'
  notes?: string | null
}

export type UpdateFeeRulePayload = Partial<CreateFeeRulePayload>

File: backend/auth-service/src/models/fee_rule.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct FeeRule {
    pub id: Uuid,
    pub group_id: Uuid,
    pub symbol: Option<String>,
    pub market: Option<String>,
    pub fee_percent: rust_decimal::Decimal,
    pub min_fee: rust_decimal::Decimal,
    pub max_fee: Option<rust_decimal::Decimal>,
    pub status: String,
    pub notes: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
    pub updated_by: Option<String>,
    pub created_by_user_id: Option<Uuid>,
}

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct FeeRuleWithGroupName {
    pub id: Uuid,
    pub group_id: Uuid,
    pub group_name: Option<String>,
    pub symbol: Option<String>,
    pub market: Option<String>,
    pub fee_percent: rust_decimal::Decimal,
    pub min_fee: rust_decimal::Decimal,
    pub max_fee: Option<rust_decimal::Decimal>,
    pub status: String,
    pub notes: Option<String>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
    pub updated_by: Option<String>,
    pub created_by_user_id: Option<Uuid>,
    pub created_by_email: Option<String>,
}
